rfmux/tools/periscope/app_runtime.py

```
@@ -1047,6 +1049,75 @@ def _start_multisweep_analysis(self, params: dict):
             print(error_msg, file=sys.stderr); traceback.print_exc(file=sys.stderr)
             QtWidgets.QMessageBox.critical(self, "Multisweep Error", error_msg)
 
+
+    async def set_nco_from_load(self, crs, nco_freq, module):
+        await crs.set_nco_frequency(nco_freq, module=module)
+        
+    
+    def _load_multisweep_analysis(self, load_params: dict):
+        """
+        Start a new multisweep analysis.
+
+        Creates a `MultisweepWindow` and a `MultisweepTask`.
+        Connects signals for progress, data updates, and completion.
+
+        Args:
+            params (dict): Parameters for the multisweep analysis, typically
+                            from a configuration dialog.
+        """
+        # MultisweepWindow from .ui, MultisweepTask from .tasks, sys, traceback from .utils
+        try:
+            if self.crs is None: QtWidgets.QMessageBox.critical(self, "Error", "CRS object not available for multisweep."); return
+            window_id = f"multisweep_window_{self.multisweep_window_count}"; self.multisweep_window_count += 1
+            params = load_params['initial_parameters']
+            target_module = params.get('module')
+            if target_module is None: QtWidgets.QMessageBox.critical(self, "Error", "Target module not specified for multisweep."); return
+            
+            dac_scales_for_window = self.dac_scales if hasattr(self, 'dac_scales') else {}
+
+            dac_scale_for_mod = load_params['dac_scales_used'][target_module]
+            dac_scale_for_board = dac_scales_for_window[target_module]
```

Won't this line fail if line 1076 triggers the "else {}" condition?

Is this just a condition that we must have that field, and so we should either (A) validate it, or (B) throw an error that the field is missing, but never guess and never fail?